src/test/scala/org/infinispan/spark/suites/StreamingSuite.scala

```diff
@@ -47,9 +47,8 @@ class StreamingSuite extends FunSuite with SparkStream with MultipleServers with
       stream.writeToInfinispan(getProperties)
 
       ssc.start()
-      ssc.awaitTerminationOrTimeout(2000)
 
-      cache.size shouldBe 3
+      waitForCondition(() => cache.size == 3)
```

Comment:
This retry logic is used in other places where using Scalatest is not convenient like on a java test ```org.infinispan.spark.JavaStreamApiTest```